Homework-№4-from-28.09.2022/task3.c

```diff
@@ -0,0 +1,205 @@
+    if (*cntOfEntries > maxSizeOfEntry) {
+        printf("File is full! Please, clear some information in it!\n");
+        return;
+    }
+    printf("Please, enter the phone number,\n");
+    printf("which is less than 100 symbols or equal to it:");
+    scanf("%s", entries[*cntOfEntries]->phoneNumber);
+    printf("Please, enter the name of the people,\n");
+    printf("which phone number you have already enter\n");
+    printf("ans which is less than 100 symbols or equal to it:");
+    scanf("%s", entries[*cntOfEntries]->name);
+    ++*cntOfEntries;
+}
+
+void command2(dataBase **entries, int *cntOfEntries) {
+    for (int i = 0; i < *cntOfEntries; ++i) {
+        printf("Phone number: %s, Name: %s\n", entries[i]->phoneNumber, entries[i]->name);
+    }
+
+}
+
+void command3(dataBase **entries, int *cntOfEntries) {
+    bool isFound = false;
+    char searchName[maxSizeOfEntry] = {0};
+    printf("Please, enter the name by which\n");
+    printf("you want to find the phone:");
+    scanf("%s", searchName);
+    for (int i = 0; i < *cntOfEntries; ++i) {
+        if (strcmp(searchName, entries[i]->name) == 0) {
+            printf("The following phone number was found: %s\n", entries[i]->phoneNumber);
+            isFound = true;
+        }
+    }
+    if (!isFound) {
+        printf("I didn't find any number by this name.\n");
+    }
+}
+
+void command4(dataBase **entries, int *cntOfEntries) {
+    bool isFound = false;
+    char searchName[maxSizeOfEntry] = {0};
+    printf("Please, enter the phone by which\n");
+    printf("you want to find the name:");
+    scanf("%s", searchName);
+    for (int i = 0; i < *cntOfEntries; ++i) {
+        if (strcmp(searchName, entries[i]->phoneNumber) == 0) {
+            printf("The following name was found: %s\n", entries[i]->name);
+            isFound = true;
+        }
+    }
+    if (!isFound) {
+        printf("I didn't find any name by this phone number.\n");
+    }
+}
+
+void freeEntries(dataBase **entries, int cntOfEntries) {
+    for (int i = 0; i < cntOfEntries; ++i) {
+        free(entries[i]);
+    }
+}
+
+void command5(dataBase **entries, int *cntOfEntries, char *nameOfFile) {
+    FILE *file = fopen(nameOfFile, "w");
+    fprintf(file, "phoneNumbers names");
+    for (int i = 0; i < *cntOfEntries; ++i) {
+        fprintf(file, "\n%s %s", entries[i]->phoneNumber, entries[i]->name);
+    }
+    fclose(file);
+}
+
+int main() {
+    char nameOfFile[maxSizeOfEntry] = {0};
+    bool firstEntry = true;
+    int cntOfEntries = 0;
+    printf("Hello! Please, enter the full file path of your file\n");
+    printf("in which we gonna input some phone numbers\n");
+    printf("and names.\n");
+    printf("I immediately warn you that the program is designed\n");
+    printf("to work with names and numbers of no more than 100 characters.\n");
+    printf("Your file path:");
+    scanf("%s", nameOfFile);
+    dataBase *entries[maxSizeOfEntry] = {NULL};
```

Comment:
```suggestion
    dataBase entries[maxSizeOfEntry] = {0};
```
Вот так сразу выделится массив со всеми записями. Ведь так это массив записей, а не массив ссылок на записи.